frontend/app/src/composables/history/filter-paginate.ts

```diff
@@ -166,9 +190,12 @@ export const useHistoryPaginationFilter = <
     if (get(userAction) && get(mainPage)) {
       // Route should only be updated on user action otherwise it messes with
       // forward navigation.
-      await router.push({
-        query: getQuery()
-      });
+      const query = getQuery();
+      if (isEqual(route.query, query)) {
+        // prevent pushing same route
+        return;
+      }
```

Comment:
in which case does this happen?

Reply:
while testing, I noticed sometimes, vue drops warning about pushing to the same route, that was why I added this extra check, but I can remove it.

Reply:
not related to hot reloading thought right?

Reply:
If this covers an actual production case then it's ok

Reply:
Ok

Reply:
on a second thought, maybe it is on hot reload...
so, I'll remove the check again, try to make a change, save and see if it happens on hot reload.